[FIX] account: missing partner bank account in vendor credit note

jpp-odoo opened the pull request:
Have a partner (vendor) with a bank account.
Create a Vendor Bills. Change the form view of the vendors bills to
display the field partner_bank_id ('Bank account').
The field shows correctly the bank account of the vendor.

Before this commit, when creating a Credit Note, the field
partner_bank_id is empty.

Now, the field parnet_bank_id is the one of the Company.

opw-1903143

oco-odoo commented:
@jpp-odoo This is probably not good. The partner_bank_id field is used to store the bank account on which we expect the payment to be received. So, in the case of a vendor bill refund, we would rather expect to see the account of the company there. 

The fact it is empty in v11 is due to this condition not taking in_refund invoices into account : https://github.com/odoo/odoo/blob/11.0/addons/account/models/account_invoice.py#L502

Whereas in 12, you can see that we do consider them
https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_invoice.py#L566

So, *maybe* it would be a good idea to consider adding in_refund in v11's condition, but you have to check that it doesn't beak everything first ;) Maybe @kebeclibre can help ;)

jpp-odoo commented:
@oco-odoo Indeed my mistake. I just added the in_refund in the condition, this only works when you create a new Vendor Credit Notes.

I also added in _prepare_refund a statement to complete the partner_bank_id field, this works in the case you clicked in 'Ask for a credit note' button.

What do you think ?

robodoo commented:
Staging failed: ci/runbot on c67a50c57fe0635628b46a18e8e2b312cd82a551 (view more at http://runbot.odoo.com/runbot/build/479655)

nim-odoo commented:
Fails on test test_vendor_bill_refund, which seems related to your modification :)

jpp-odoo commented:
robodoo retry

robodoo commented:
Staging failed: ci/runbot on f931042a7a94e624738e5a865e9567b633069955 (view more at http://runbot.odoo.com/runbot/build/481141)

jpp-odoo commented:
robodoo retry

robodoo commented:
I'm sorry, @jpp-odoo. Retry makes no sense when the PR is not in error.

nim-odoo commented:
robodoo r+

robodoo commented:
Merged, thanks!

mreficent commented on addons/account/models/account_invoice.py:
```diff
@@ -499,16 +499,19 @@ def default_get(self,default_fields):
+    def _get_partner_bank_id(self, company_id):
+        company = self.env['res.company'].browse(company_id)
+        if company.partner_id:
+            return self.env['res.partner.bank'].search([('partner_id', '=', company.partner_id.id)], limit=1)
```
but what returns if not company.partner_id? It should return at least None or self.env['res.partner.bank'].

smetl replied:
As `partner_id` is a required field of `res.partner.bank`, the result will be an empty recordset.
